code/game/machinery/cryo.dm

```diff
@@ -225,47 +318,72 @@
 				var/heal_brute = occupant.getBruteLoss() ? min(1, 20/occupant.getBruteLoss()) : 0
 				var/heal_fire = occupant.getFireLoss() ? min(1, 20/occupant.getFireLoss()) : 0
 				occupant.heal_limb_damage(heal_brute,heal_fire)
+				//to_chat(world, "<span class='notice'>DEBUG: Process_Occupant. Occupant: [occupant]. Brute Heal: [heal_brute] Burn Heal: [heal_fire]</span>")
 		var/has_cryo = occupant.reagents.get_reagent_amount("cryoxadone") >= 1
 		var/has_clonexa = occupant.reagents.get_reagent_amount("clonexadone") >= 1
 		var/has_cryo_medicine = has_cryo || has_clonexa
+		//to_chat(world, "<span class='notice'>DEBUG: Process_Occupant. Occupant: [occupant]. Cryo Medicine?: [has_cryo_medicine]</span>")
 		if(beaker && !has_cryo_medicine)
 			beaker.reagents.trans_to(occupant, 1, 10)
 			beaker.reagents.reaction(occupant)
 
 
 
-/obj/machinery/atmospherics/unary/cryo_cell/proc/go_out()
+/obj/machinery/cryo_cell/proc/go_out(auto_eject = null, dead = null)
 	if(!( occupant ))
 		return
-	//for(var/obj/O in src)
-	//	O.loc = loc
 	if (occupant.client)
 		occupant.client.eye = occupant.client.mob
 		occupant.client.perspective = MOB_PERSPECTIVE
 	occupant.loc = get_step(loc, SOUTH)	//this doesn't account for walls or anything, but i don't forsee that being a problem.
 	if (occupant.bodytemperature < 261 && occupant.bodytemperature >= 70) //Patch by Aranclanos to stop people from taking burn damage after being ejected
 		occupant.bodytemperature = 261									  // Changed to 70 from 140 by Zuhayr due to reoccurance of bug.
-//	occupant.metabslow = 0
+	if(auto_eject) //Turn off and announce if auto-ejected because patient is recovered or dead.
+		turn_off()
+		if(release_notice) //If auto-release notices are on as it should be, let the doctors know what's up
+			playsound(src.loc, 'sound/machines/ping.ogg', 100, 14)
+			var/reason = "Reason for release:</b> Patient recovery."
+			if(dead)
+				reason = "<b>Reason for release:</b> Patient death."
+			var/mob/living/silicon/ai/AI = new/mob/living/silicon/ai(src, null, null, 1)
+			AI.SetName("Cryotube Notification System")
+			AI.aiRadio.talk_into(AI,"Patient [occupant] has been automatically released from [src] at: [get_area(occupant)]. [reason]","MedSci","announces")
+			cdel(AI)
 	occupant = null
 	update_use_power(1)
 	update_icon()
 	return
-/obj/machinery/atmospherics/unary/cryo_cell/proc/put_mob(mob/living/carbon/M as mob)
+
+/obj/machinery/cryo_cell/proc/turn_off()
+	on = FALSE
+	stop_processing()
+	update_icon()
+
+/obj/machinery/cryo_cell/proc/turn_on()
 	if (stat & (NOPOWER|BROKEN))
 		to_chat(usr, "\red The cryo cell is not functioning.")
 		return
-	if (!istype(M))
-		to_chat(usr, "\red <B>The cryo cell cannot handle such a lifeform!</B>")
+	on = TRUE
+	start_processing()
```

Comment:
im not sure if this being moved here is going to be super useful, @ninjanomnom do you think this is better than putting it in New() or what?

Reply:
Yeah it's better that things only process when they actually have something to do

Reply:
kk